### PR TITLE
Run AutoGPT without GUI

### PR DESCRIPTION
**Pull request by Zep4yrWang**
### Background

Current AutoGPT cannot be run on servers or computing clusters which typically cannot open browsers in GUI mode. 

### Changes

autogpt/args.py : added parsing for '--no-gui'
autogpt/config/config.py: added default value for args.py
autogpt/commands/web_selenium: options to disable browser window. 

### Documentation

Now supports running selenium without GUI, enabling running on terminal-only machines. This function is enabled by
`python -m autogpt --no-gui`

### Test Plan

Tested on cloud servers and worked. 

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by SargeDGz** [comment]
my work around was to install a desktop environment and rdp in and then run autogpt but yea rdp over internet is bad, thanks for this hope it gets merged in to the project.

**Comment by yudikubota** [comment]
I tried running it and got the following error:

image

**Comment by Zep4yrWang** [comment]
> I tried running it and got the following error:
> 
> image

Hi. 'master' branch is being updated very frequently and sometimes requirements.txt falls behind. Manual installation will solve this. 
`pip install GitPython`

**Review by stellarkey — COMMENTED** [review]
A good solution to non-GUI-able environments!

**Review by stellarkey — APPROVED** [review]
I think it's approvable.

File: autogpt/config/config.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ def __init__(self) -> None:
         self.continuous_limit = 0
         self.speak_mode = False
         self.skip_reprompt = False
+        self.no_gui=False
```
Comment (stellarkey) [inline autogpt/config/config.py:27]: please make it `self.no_gui = False`

**Review by stellarkey — CHANGES_REQUESTED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by anonhostpi** [comment]
Recommend building an asynchronous API to talk to the currently running process would be the best method.

Some kind of IPC. I'm sure there are several IPC libraries for Python. Another option would be to host a local http server that takes REST/GRAPHQL calls.

**Comment by Boostrix** [comment]
especially once multiple async agents are becoming a thing (#765, #2422), this will be highly useful

**Comment by Pwuts** [comment]
Since #1473+#2816, browsers open in headless mode by default. This can be adjusted using an environment variable or the `.env` config, which is probably enough for now. Still, thanks for submitting!

**Pull request closed without merging**